### PR TITLE
feature(nemesis): Change TWCS unit size and check reshaping

### PR DESCRIPTION
**Pull request by aleksbykov**
Trello: https://trello.com/c/5OpkxxlQ

Changing TWCS window size with next reshape
could bring significant write amplicification.
New nemesis will change the window size and
restart the node

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/nemesis.py
Comment (KnifeyMoloko) [inline sdcm/nemesis.py:2100]: Don't we need a bit of time for the reshape to actually happen between Scylla restarting and getting this list?

**Review by KnifeyMoloko — CHANGES_REQUESTED** [review]
One small question, otherwise LGTM

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:2100]: It happened during scylla load, so once scylla has been started, it already should be in log. Another possible issue here, is that autocompaction could compact sstables and segregate them while scylla is going to be stopped.

**Comment by roydahan** [comment]
Shouldn't be backport to 5.0 and 2022.1 at this stage because it will change the nemesis_seed order.
For 5.0 and/or 2022.1 we should decide how/where to run it.

File: sdcm/nemesis.py
Comment (roydahan) [inline sdcm/nemesis.py:1506]: Let's use the nemesis_seed here for selection (or any other seed) to make sure tests are repeatable.

File: sdcm/nemesis.py
Comment (roydahan) [inline sdcm/nemesis.py:2080]: same with seed.

File: sdcm/nemesis.py
Comment (roydahan) [inline sdcm/nemesis.py:2092]: Do we want to have a sleep here for a while?
Maybe to have some of the sstables to be written with the new properties, so there will be a mixture of the old sstables and the new sstables.
If you do wait, let's make it as a variable based on test duration or something like that.

File: sdcm/nemesis.py
Comment (roydahan) [inline sdcm/nemesis.py:2104]: So even if we don't see reshapes it's fine?
What about the write_amplification? Is there a way to check that it happened?

File: sdcm/nemesis.py
Comment (roydahan) [inline sdcm/nemesis.py:2111]: Maybe we need to set the values back to what they were before the change?

**Review by roydahan — CHANGES_REQUESTED** [review]
See my comments below.

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:1506]: done

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:2080]: done

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:2092]: Added

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:2104]: > So even if we don't see reshapes it's fine? 

Yes, it means, that autocompaction is already segregate sstables by timewindows

> What about the write_amplification? Is there a way to check that it happened?

I am not sure it is possible  to do, while traffic is running continuously. if scylla have  some metric for that?

Comment (aleksbykov) [reply sdcm/nemesis.py:2104]: I think the best way is to  have some performance tests for that

File: sdcm/nemesis.py
Comment (aleksbykov) [reply sdcm/nemesis.py:2363]: I asked Raphael about this operation is it support or not.  if it will be ok, i will add

File: sdcm/nemesis.py
Comment (roydahan) [reply sdcm/nemesis.py:1696]: where?

**Comment by roydahan** [comment]
Let's wait for the run without any nemesis here https://jenkins.scylladb.com/view/scylla-5.0/job/scylla-5.0/job/SCT_Reproducers/job/longevity-twcs-48h-test-NoMonkey/

And let's check the number of sstables we have during the run.

then, we need to run a test with this nemesis only and compare the differences.

**Comment by roydahan** [comment]
For now, timeseries in scylla-bench is suspected as problematic.
Holding this PR till we clear it.

File: test-cases/longevity/longevity-twcs-3h.yaml
Comment (roydahan) [inline test-cases/longevity/longevity-twcs-3h.yaml:4]: I think that for timeseries, the typical use case won't have such a big row count.
Maybe we misinterpret the scylla-bench usage and we need to keep the partitions small (for example 10 rows that are reporting metrics of a sensor in particular time).

**Comment by roydahan** [comment]
Please remove the commit that changes the yaml and resolve conflicts.
Then, we will merge it.

**Comment by aleksbykov** [comment]
- [ ] remove 'revert' commits
- [ ] resolve conflicts

**Review by roydahan — DISMISSED** [review]

**Comment by roydahan** [comment]
Please check why some checks are failing

**Comment by aleksbykov** [comment]
@roydahan , all test passed.

**Review by fruch — APPROVED** [review]

File: sdcm/nemesis.py
Comment (fruch) [inline sdcm/nemesis.py:2438]: @aleksbykov can you explain again this calculation ?

test_duration is in minutes, while `sleep_timeout` is in seconds

so for a 3 day test:
4320 * 0.02 = 86.4

for a 4h test:
360 * 0.02 = 7.2

so for one test it would sleep for a bit more then minute, and for the other it would sleep 7 sec, I fail to understand how the duration play any part of the TWCS configuration, and why the timing here isn't base on the selected table configuration...

**Pull request merged**
